src/quo2/components/tabs/tabs.cljs

```diff
@@ -1,21 +1,138 @@
          (for [{:keys [label id]} data]
            ^{:key id}
-           [rn/view {:margin-right (if (= size 32) 12 8)}
+           [rn/view {:margin-right (if (= size default-tab-size) 12 8)}
             [tab/tab
-             {:id id
-              :size size
-              :active (= id active-id)
-              :on-press #(do (reset! active-tab-id %)
-                             (when on-change
-                               (on-change %)))}
-             label]])]))))
\ No newline at end of file
+             {:id       id
+              :size     size
+              :active   (= id active-id)
+              :on-press (fn [^js press-event id]
+                          (reset! active-tab-id id)
+                          (when on-change
+                            (on-change press-event id)))}
+             label]])]))))
+
+(defn- calculate-fade-end-percentage
+  [{:keys [offset-x content-width layout-width max-fade-percentage]}]
+  (let [fade-percentage (max max-fade-percentage
+                             (/ (+ layout-width offset-x)
+                                content-width))]
+    ;; Truncate to avoid unnecessary rendering.
+    (if (> fade-percentage 0.99)
+      0.99
+      (number-utils/naive-round fade-percentage 2))))
+
+(defn scrollable-tabs
+  "Just like the component `tabs`, displays horizontally scrollable tabs with
+  extra options to control if/how the end of the scroll view fades.
+
+  Tabs are rendered using ReactNative's FlatList, which offers the convenient
+  `scrollToIndex` method. FlatList accepts VirtualizedList and ScrollView props,
+  and so does this component.
+
+  Usage:
+  [tabs/scrollable-tabs
+    {:scroll-on-press? true
+     :fade-end?        true
+     :on-change        #(...)
+     :default-active   :tab-a
+     :data             [{:id :tab-a :label \"Tab A\"}
+                        {:id :tab-b :label \"Tab B\"}]}]]
+
+  Opts:
+  - `size` number
+  - `scroll-on-press?` When non-nil, clicking on a tab centers it the middle
+  (with animation enabled).
+  - `fade-end?` When non-nil, causes the end of the scrollable view to fade out.
+  - `fade-end-percentage` Percentage where fading starts relative to the total
+  layout width of the `flat-list` data."
+  [{:keys [default-active fade-end-percentage]
+    :or   {fade-end-percentage 0.8}}]
+  (let [active-tab-id (reagent/atom default-active)
+        fading        (reagent/atom {:fade-end-percentage fade-end-percentage})
+        flat-list-ref (atom nil)]
+    (fn [{:keys [data
+                 fade-end-percentage
+                 fade-end?
+                 on-change
+                 on-scroll
+                 scroll-event-throttle
+                 scroll-on-press?
+                 size]
+          :or   {fade-end-percentage   fade-end-percentage
+                 fade-end?             false
+                 scroll-event-throttle 64
+                 scroll-on-press?      false
+                 size                  default-tab-size}
```

Comment:
I think @flexsurfer recommends keeping default props to the screens namespace .. But IDK since this component doesn't have a screens namespace.
And ideas Andre?

Reply:
Interesting, but then doesn't it cause a circular dependency? E.g. `screens` requires `tabs`, but `tabs` also requires `screens` to get the default map. I still find it weird to couple different namespaces for default function maps, even more because the screens namespace is not production code, it's like test code.

Reply:
You're right Icaro, It's just code for manual visual testing
But I don't think `tabs` should require `screens.tabs`